File: backend/src/main/java/com/documed/backend/users/Subscription.java
package com.documed.backend.users;

import java.math.BigDecimal;
import java.util.List;
import lombok.Data;
import lombok.NonNull;

@Data
public class Subscription {

  private final int id;
  @NonNull private String name;
  @NonNull private BigDecimal price;
  private List<User> users;
  private List<SubscriptionService> subscriptionServices;
}

File: frontend/src/shared/types/Subscription.ts
import { SubscriptionServiceLite } from './SubscriptionService';
import { UserLite } from './User';

export interface Subscription {
  users: UserLite[] | null;
  subscriptionServices: SubscriptionServiceLite[] | null;
}

export interface SubscriptionLite {
  id: number;
  name: string;
  price: number;
}
